client/x/evmengine/keeper/abci.go

```diff
@@ -53,10 +53,20 @@ func (k *Keeper) PrepareProposal(ctx sdk.Context, req *abci.RequestPreparePropos
 
 	appHash := common.BytesToHash(ctx.BlockHeader().AppHash)
 
-	withdrawals, err := k.evmstakingKeeper.PeekEligibleWithdrawals(ctx)
+	maxWithdrawals, err := k.evmstakingKeeper.MaxWithdrawalPerBlock(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "get max withdrawal per block")
+	}
+	withdrawals, err := k.evmstakingKeeper.PeekEligibleWithdrawals(ctx, int(maxWithdrawals))
 	if err != nil {
 		return nil, errors.Wrap(err, "error on withdrawals dequeue")
 	}
+	maxRewardWithdrawals := int(maxWithdrawals) - len(withdrawals)
+	rewardWithdrawals, err := k.evmstakingKeeper.PeekEligibleRewardWithdrawals(ctx, maxRewardWithdrawals)
```

Comment:
If we move to an epoch-based withdrawal system later, it could be possible to flood the stake withdrawal queue s.t. `maxRewardWithdrawals = 0` for many epochs, which delays reward receiving and could affect reward assumptions, e.g. restaking protocols. Thus, an option to consider is whether making `maxRewardWithdrawals` dependent on `maxWithdrawals` (ie. maxStakeWithdrawals) or not. cc @0xHansLee @edisonz0718

Reply:
That's true. We may need other approach to withdraw these when epoch system is introduced.